### PR TITLE
fix: Make Git conflict markers check more precise

### PR DESCRIPTION
**Pull request by yermulnik**
The https://github.com/super-linter/super-linter/pull/6113 introduced new Git merge conflicts linter check, that is error prone in some conditions (see screenshot).

This PR adjusts this check to be more precise and exact in a way that all three markers should be found to identify check as failed. Additionally improve Git merge conflict markers matchers.

Additionally update one good test file.

Repro:
image
```shell
> grep -E '^(<<<<<<<|=======|>>>>>>>)' README.md
==========================================================
=============   MegaLinter, by OX.security   =============
=========  https://ox.security?ref=megalinter  ===========
==========================================================
```

## Readiness checklist

In order to have this pull request merged, complete the following tasks.

### Pull request author tasks

- [x] I checked that all workflows return a success.
- [x] I included all the needed documentation for this change.
- [x] I provided the necessary tests.
- [x] I squashed all the commits into a single commit.
- [x] I followed the
      [Conventional Commit v1.0.0 spec](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I wrote the necessary upgrade instructions in the
      [upgrade guide](../docs/upgrade-guide.md).
- [ ] If this pull request is about and existing issue, I added the
      `Fix #ISSUE_NUMBER` or `Close #ISSUE_NUMBER` text to the description of
      the pull request.

### Super-linter maintainer tasks

- [x] Label as `breaking` if this change breaks compatibility with the previous
      released version.
- [x] Label as either: `automation`, `bug`, `documentation`, `enhancement`,
      `infrastructure`.
- [ ] Add the pull request to a milestone, eventually creating one, that matches
      with the version that release-please proposes in the
      `preview-release-notes` CI job.

### CODE REVIEW

File: test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt [hunk 1]
```diff
@@ -1 +1,5 @@
+<<<<<<<<<<<<<
```
[the remark this answers is not in the document]
Comment (yermulnik) [reply test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt:1]: This file does not contain merge conflict markers. It contains random strings that look like a merge conflict markers. You may validate by feeding this file to the script:
```shell
> scripts/git-merge-conflict-markers.sh test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt
No merge conflicts found in test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt
```

File: scripts/git-merge-conflict-markers.sh [hunk 2]
```diff
@@ -4,15 +4,27 @@ set -o errexit
...
   echo "1.0.0"
   exit 0
 fi
 
-if grep -l -E "${GIT_MERGE_CONFLICT_EXPRESSION}" "$@"; then
-  echo "Found Git merge conflict markers"
+declare -i errors=0
+
+for file in "$@"; do
+  if grep -q -E "${GIT_MERGE_CONFLICT_START}" "$file" &&
```
[the remark this answers is not in the document]
Comment (yermulnik) [reply scripts/git-merge-conflict-markers.sh:19]: This is done by means of update to `test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt` file.

File: test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (yermulnik) [reply test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt:1]:
```shell
> scripts/git-merge-conflict-markers.sh test/linters/git_merge_conflict_markers/git_merge_conflict_markers_*
Found Git merge conflict markers: "test/linters/git_merge_conflict_markers/git_merge_conflict_markers_bad_01.txt"
Found Git merge conflict markers: "test/linters/git_merge_conflict_markers/git_merge_conflict_markers_bad_02.txt"
```

[the remark this answers is not in the document]
Comment (yermulnik) [reply test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt:1]: Please let me know if there's still a misunderstanding here. Thanks.

**Comment by yermulnik** [comment]
Dang, I can't seem to figure out how to squash that merge commit. Anything I do usually to squash commits just brings this back to having two commits 😲 Any pointers?

**Comment by claudioandre-br** [comment]
I would like to ask/propose that you guys add version 7.2.1 as a milestone for this PR.

File: test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ferrarimarco) [reply test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt:1]: Got it. Can you add this explanation in the test file itself, so we're not confused? Explaining how this looks like a conflict marker, but it's not will help clarify.

File: scripts/git-merge-conflict-markers.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (ferrarimarco) [reply scripts/git-merge-conflict-markers.sh:19]: Ok

**Comment by ferrarimarco** [comment]
> Dang, I can't seem to figure out how to squash that merge commit. Anything I do usually to squash commits just brings this back to having two commits 😲 Any pointers?

This might be because you're using the `main` branch in your fork directly.

I would do this:

1. Remove the merge commit from the `main` branch on your fork
2. Move your commit to a dedicated branch
3. Sync your `main` branch with the upstream `main` branch
4. Rebase the dedicated branch on top of your main branch
5. Update this PR to pull from the dedicated branch, instead of the `main` branch

Note that if there are automatically fixable linting errors, there's a workflow to automatically fix and commit updates.

Hope this helps!

File: test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (yermulnik) [reply test/linters/git_merge_conflict_markers/git_merge_conflict_markers_good_02.txt:1]: 👌🏻

**Comment by yermulnik** [comment]
> Hope this helps!

Argh, still can't figure this out. Let me close this PR and go with a corrected one.

**Comment by yermulnik** [comment]
https://github.com/super-linter/super-linter/pull/6379

**Pull request closed without merging**